src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -1510,6 +1512,13 @@ void GcInfoDecoder::ReportRegisterToGC(  // AMD64
 {
     GCINFODECODER_CONTRACT;
 
+#ifndef NATIVEAOT
```

Comment:
I would not bother with ifdefing tiny bits of code like this for native AOT.

Reply:
If this is fixing build break, I would rather define a dummy `ReportFPBasedSlotsOnly` in native AOT. It can be defined to be `0` that should allow the compiler to eliminate this branch for native AOT.

Reply:
This was just for my testing purposes, I didn't want to merge that in.

Reply:
Actually, somehow CI has screwed this PR. For some reason, the description of the change I've added is missing and the commit doesn't contain the cleanup stuff. I need to check what happened and re-commit what I have locally.

Reply:
Ah sorry, I was commenting on the log above. This is obviously needed. Yes, this was here to fix NativeAOT build break. I'll modify it the way you've suggested.